server/api.go

```diff
@@ -189,13 +189,23 @@ func WriteString(w http.ResponseWriter, str string) {
 //  - SessionRequest instances (DisclosureRequest, SignatureRequest, IssuanceRequest)
 //  - JSON representations ([]byte or string) of any of the above.
 func ParseSessionRequest(request interface{}) (irma.RequestorRequest, error) {
+	rr, e := ParseInput(request)
+	if e != nil {
+		return nil, e
+	}
+	rr.Base().SetDefaultsIfNecessary()
+
+	return rr, e
+}
+
+func ParseInput(request interface{}) (irma.RequestorRequest, error) {
```

Comment:
Since this function is only used here and additionally not necessarily suitable for use by others, I recommend to not export it (i.e. rename to `parseInput`).

Reply:
done